web/src/main/java/com/hendzior/web/controller/AnimalController.java

```diff
@@ -0,0 +1,84 @@
+package com.hendzior.web.controller;
+
+import com.hendzior.core.DataInputValidator;
+import com.hendzior.core.dao.AnimalDataAccess;
+import com.hendzior.core.dao.CustomerDataAccess;
+import com.hendzior.core.model.Animal;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Slf4j
+@RequestMapping("/animals")
+@Controller
+public class AnimalController {
+
+    private AnimalDataAccess animalDataAccess;
+    private CustomerDataAccess customerDataAccess;
+    private DataInputValidator dataInputValidator;
+
+    public AnimalController(AnimalDataAccess animalDataAccess, CustomerDataAccess customerDataAccess, DataInputValidator dataInputValidator) {
+        this.animalDataAccess = animalDataAccess;
+        this.customerDataAccess = customerDataAccess;
+        this.dataInputValidator = dataInputValidator;
+    }
+
+    @GetMapping
+    public String getAnimals(ModelMap modelMap) {
+        modelMap.addAttribute("animals", animalDataAccess.findAll());
+
+        return "show";
+    }
+
+    @GetMapping("/add")
+    public String createAnimal(ModelMap modelMap) {
+        if (!modelMap.containsAttribute("animal")) {
+            modelMap.addAttribute("animal", new Animal());
+        }
+        modelMap.addAttribute("customers", customerDataAccess.findAll());
+        return "addanimal";
+    }
+
+    @PostMapping
+    public String saveAnimal(@ModelAttribute @Valid Animal animal, BindingResult bindingResult, ModelMap modelMap, RedirectAttributes redirectAttributes) {
+      if(bindingResult.hasErrors()){
+          redirectAttributes.addFlashAttribute("org.springframework.validation.BindingResult.animal", bindingResult);
+          redirectAttributes.addFlashAttribute("animal", animal);
+          return "redirect:/animals/add";
+      }
+        animalDataAccess.save(animal);
+        modelMap.addAttribute("message", "animal : " + animal.getName() + " added!");
```

Comment:
Would be better to set flag: animalAdded = true and show proper message based on that value.

Reply:
I dont get it, can you give me an example ?